    /**
     * test that Error types (BoolErrRecord) are supported properly.
     * @see #testBool
     */
    @Test
    public void testErr() throws IOException {

        try (Workbook wb1 = _testDataProvider.createWorkbook()) {
            Sheet s = wb1.createSheet("testSheet1");
            Row r;
            Cell c;

            // B1
            r = s.createRow(0);
            c = r.createCell(1);
            assertEquals(0, c.getRowIndex());
            assertEquals(1, c.getColumnIndex());
            c.setCellErrorValue(FormulaError.NULL.getCode());
            assertEquals("B1 value == #NULL!", FormulaError.NULL.getCode(), c.getErrorCellValue());

            // C1
            c = r.createCell(2);
            assertEquals(0, c.getRowIndex());
            assertEquals(2, c.getColumnIndex());
            c.setCellErrorValue(FormulaError.DIV0.getCode());
            assertEquals("C1 value == #DIV/0!", FormulaError.DIV0.getCode(), c.getErrorCellValue());

            try (Workbook wb2 = _testDataProvider.writeOutAndReadBack(wb1)) {
                s = wb2.getSheet("testSheet1");

                r = s.getRow(0);
                assertEquals("Row 1 should have 2 cells", 2, r.getPhysicalNumberOfCells());

                c = r.getCell(1);
                assertEquals(0, c.getRowIndex());
                assertEquals(1, c.getColumnIndex());
                assertEquals(CellType.ERROR, c.getCellType());
                assertEquals("B1 value == #NULL!", FormulaError.NULL.getCode(), c.getErrorCellValue());

                c = r.getCell(2);
                assertEquals(0, c.getRowIndex());
                assertEquals(2, c.getColumnIndex());
                assertEquals(CellType.ERROR, c.getCellType());
                assertEquals("C1 value == #DIV/0!", FormulaError.DIV0.getCode(), c.getErrorCellValue());
            }
        }
    }

